studio-docs/source/sharing-graphs.mdx

```diff
@@ -0,0 +1,191 @@
+```
+
+</MultiCodeBlock>
+
+The page begins listening for `message` events, and it calls the `onPostMessageReceived` callback whenever one occurs. Now, we need to define that callback:
+
+<MultiCodeBlock>
+
+```ts:title=explorer-listener.ts
+// This is the name prefix for relevant messages posted by the iframe
+const EXPLORER_QUERY_MUTATION_REQUEST = "ExplorerRequest";
+
+// Callback definition
+const onPostMessageReceived = (event:MessageEvent<{
+  name?: string,
+  operation?: string,
+  operationName?: string,
+  variables?: string,
+  headers?: Record<string, string>
+}>) => {
+
+  // Make sure the message's name starts with `ExplorerRequest`
+  const isQueryOrMutation = 'name' in event.data && event.data.name?.startsWith(`${EXPLORER_QUERY_MUTATION_REQUEST}:`);
+
+  if (isQueryOrMutation && event.data.name && event.data.operation) {
+
+    // Extract the operation ID from the message name
+    const operationId = event.data.name.split(":")[1];
+    // Identify the embedded Explorer iframe
+    const embeddedExplorerIFrame = document.getElementById("embedded-explorer") as HTMLIFrameElement ?? undefined;
+    const { operation, operationName, variables, headers } = event.data;
+    // Execute the corresponding GraphQL operation via fetch (shown below)
+    executeOperation({
+      operation: event.data.operation,
+      operationName,
+      variables,
+      headers,
+      embeddedExplorerIFrame,
+      operationId,
+    });
+  }
+};
+```
+
+</MultiCodeBlock>
+
+As shown above, when a user initiates a query or mutation operation, the Explorer `iframe` posts a message with a name that starts with the string `ExplorerRequest:`. Whenever we receive a message with a matching name, we extract the operation's details from the event and use them to execute the operation.
+
+#### 2. Execute the operation
+
+Our `executeOperation` method looks like this:
+
+<MultiCodeBlock>
+
+```ts:title=explorer-listener.ts
+// This is the name prefix for response messages expected by the iframe
+const EXPLORER_QUERY_MUTATION_RESPONSE = "ExplorerResponse";
+
+// Replace this with your public variant's GraphQL endpoint URL!
+const GRAPHQL_ENDPOINT = "https://apollo-fullstack-tutorial.herokuapp.com/";
+
+// Replace the value of this URL's `graphRef` query parameter
+// with your public variant's graph ref
+const EMBEDDABLE_EXPLORER_URL = "https://explorer.embed.apollographql.com/?graphRef=Apollo-Fullstack-Demo-o3tsz8@current";
+
+async function executeOperation({
+  operation,
+  operationName,
+  variables,
+  headers,
+  embeddedExplorerIFrame,
+  operationId,
+}: {
+  operation: string,
+  operationName?: string,
+  variables?: JSONValue,
+  headers?: Record<string, string>,
+  embeddedExplorerIFrame?: HTMLIFrameElement,
+  operationId: string
+}) {
+  // Send the operation to the GraphQL server via fetch
+  const response = await fetch(
+    GRAPHQL_ENDPOINT,
+    {
+      method: "POST",
+      headers: getHeadersWithContentType(headers),
```

Comment:
this function will need to be defined. It is just a function that adds 'content-type': 'application/json' if it isn't already there (case insensitive)